content/process/git-flow.md

```diff
@@ -1,3 +1,95 @@
 # Git
 
-TODO
+This guide assumes that you are already familiar with the basic functionalities of `git`:
+
+* Cloning repositories
+* Checking branches out
+* Committing changes
+* Pushing
+* Pulling
+
+And it will walk you through slightly more advanced concepts that will come in handy when working on Optakt projects.
+
+## Rebasing
+
+The most important thing you'll likely need to master is _rebasing_.
+Whenever the base branch one of your pull requests is against gets updated with changes that conflict with yours, you will need to rebase your branch before it can be properly reviewed and merged.
+
+!!! warning "git merge"
+
+    We do not use `git merge` at Optakt, because it would result in a messier commit history.
+    Rebasing, and squashing commits before merging pull requests ensures that when looking at master, each commit corresponds to a single pull request.
+
+1. Fetch the latest version of the base branch you need to rebase against. In most cases this will be `master`.
+2. Run `git rebase <base branch>` from your local branch.
+3. This will make `git` rewind history and put your HEAD back to the first commit that diverges between the two branches.
+4. One by one, commits will be automatically applied to your local branch until one with conflicts is reached.
+5. You will now need to resolve the conflicts
+   1. This can be done with a GUI using most development environments, or manually by running `git status` to see which files have conflicts that need to be resolved and modifying them manually.
+6. Now, run `git rebase --continue` to continue rebasing.
+7. Once the final commit is reached, you can run `git push --force` to overwrite the previous version of your branch with the new rebased one.
+
+??? hint "If you made a mistake"
+
+    If you made a mistake and overwrote your remote and local branches with one that does not work, no worries!
+    You can always use [`git reflog`](#git-reflog) to get back to the previous state of your branch, before you started rebasing.
+
+For more details on `git  rebase`, please read the [git manual](https://git-scm.com/book/en/v2/Git-Branching-Rebasing).
+
+### Alternative to Rebasing
+
+In some cases, if the base branch was modified by force and now has tons of conflicts that are not related to your own changes, you might be better off not rebasing, so that you do not have to deal with resolving many conflicts you are unsure of.
+
+In that situation, you can follow these simple steps:
+
+1. `git log` on your branch to know the hashes of your commits
+2. `git checkout <base branch>`
+3. `git pull`
+4. `git checkout -b tmp`
+5. `git cherry-pick <commit hash>` for each of your commits
```

Comment:
Perhaps a better course of action would be to first do an interactive rebase and squash all commits for the original branch to a single one. Then when applying a cherry-pick there's a single commit to apply and the conflict will be resolved only once.

Reply:
But this doesn't preserve commit history, which is one of the things I'm trying to keep neat and consistent through the use of this guide :p 

I mean we can then split the commits again afterwards but that seems like a lot of extra work to me. WDYT?

Reply:
That's true, the commit history is lost. I just mentioned it since sometimes it's tedious to resolve conflicts for a number of commits, so it's easier. Probably better to not have it as standard practice.

Reply:
I think in the end that the method I'm suggesting here is just as fast (as long as you dont have 150 commits on your branch of course 😬) and preserves the history, so it might be better in the guide IMO :p 

I'm sure there is a better way than both of our methods that we don't know of though. We should try to hire a git-bender one day 😄 They exist and it's insane what they do